beep/structure.py

```diff
@@ -1073,16 +1081,31 @@ def get_maccor_quantity_sum(data, quantity, state_type):
 
         """
         state_code = MACCOR_CONFIG["{}_state_code".format(state_type)]
-        quantity_agg = data["_" + quantity].where(
-            data["_state"] == state_code, other=0.0, axis=0
-        )
+        quantity_agg = data['_' + quantity].where(data["_state"] == state_code, other=0, axis=0)
+
+        # If a waveform step is present, maccor initializes waveform-specific quantities
+        # that are to be used in place of '_capacity' and '_energy'
+
+        if data['_wf_chg_cap'].notna().sum():
```

Comment:
I like this. Since it's a method that is specific to maccor already, and we get to use the "built-in" functionality of the system